src/api/AutoBundler.ts

```diff
@@ -81,6 +81,22 @@ export class AutoBundler {
     if (this.fs.existsSync(this.packageJsonFile)) {
       this.processPackageJson(params);
     }
+    else {
+      // If we get here then we've not found a package.json file. it remains possible
+      // that sufficient command line parameters have been set to allow a NODEJSAPP
+      // to be created. If so, process them.
+      if (this.startscriptOverride !== undefined ||
+          this.nodejsappOverride   !== undefined ||
+          this.portOverride        !== undefined ) {
+        this.bundle.addNodejsappDefinition(this.nodejsappOverride, this.startscriptOverride, this.portOverride);
+        try {
+          params.response.console.log('NODEJSAPP "' + this.nodejsappOverride + '" defined for startscript "' + this.startscriptOverride + '"');
+        }
+        catch (error) {
+          // logging errors can be thrown in some of the mocked tests... just ignore it.
+        }
+      }
+    }
```

Comment:
What happens if sufficient parameters aren't provided here? I think it would be better to fail with a message.

Reply:
Those are logical ORs, not ANDs (at least they're supposed to be, they would be in Java!), so if any of those three parameters are set then an attempt is made to create a NODEJSAPP, and if any of the values aren't provided then the existing validation should catch that.